### ISSUE TITLE
allow different mappings per chain

### ISSUE THREAD

**Issue by aspiers** [aspiers/rolod0x#23]
Due to [the way that contract addresses are calculated by the `CREATE` opcode](https://ethereum.stackexchange.com/questions/760/how-is-the-address-of-an-ethereum-contract-computed), it's possible for the same contract address to be used across multiple chains for deployment of entirely different contract code and hence purposes.

Similarly, a user might choose to use an EoA on one chain for an entirely different purpose to how they use it on another chain.

Therefore, there is value in rolod0x optionally supporting different address -> name mappings per chain.

Unfortunately there is no general solution for determining the context of which chain is being referred to when an address is mentioned on a web page.  However there are plenty of common use cases where it is easy to determine the chain simply from the URL, e.g. most block explorers have a unique DNS domain per chain like `mumbai.polygonscan.com`, and Gnosis Safe includes the network as a prefix of the URL's first path segment:

- `https://app.safe.global/eth:0x.../home`
- `https://app.safe.global/matic:0x.../home`

So we should add some intelligence which figures out the web page's `chainId` where possible, and then supports chain-specific mappings.

**Comment by aspiers** [comment]
Worth noting that the [tokenlists.org](https://tokenlists.org) JSON schema includes `chainId`:

https://github.com/Uniswap/token-lists/blob/423b5a81297fe8b7d9f864e80d999ea4ace83f0a/src/tokenlist.schema.json#L205-L213

**Comment by aspiers** [comment]
I think `chainId` makes sense, since it can be cross-referenced against https://chainid.network/chains.json

**Comment by RyanCwynar** [comment]
yeah, that works. and maybe if it were a comma delimited list... 
default to all for EOAs or `*` if stated explicitly or not blank
`chainId` for specific chain or `1,137` for something specific like mainnet and polygon

**Comment by aspiers** [comment]
I've just filed #244 which is now the ideal place to discuss a general address book schema.
